adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -847,6 +942,51 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 				IdleTimeout:       ptypes.DurationProto(time.Duration(config.Envoy.Upstream.Timeouts.RouteIdleTimeoutInSeconds) * time.Second),
 			},
 		}
+		if config.Envoy.RateLimit.Enabled && ratelimitLevel != "" {
+			basePath += resourcePathParam
+			rateLimit := routev3.RateLimit{
+				Actions: []*routev3.RateLimit_Action{
+					{
+						ActionSpecifier: &routev3.RateLimit_Action_GenericKey_{
+							GenericKey: &routev3.RateLimit_Action_GenericKey{
+								DescriptorKey:   "org",
+								DescriptorValue: params.organizationID,
+							},
+						},
+					},
+					{
+						ActionSpecifier: &routev3.RateLimit_Action_GenericKey_{
+							GenericKey: &routev3.RateLimit_Action_GenericKey{
+								DescriptorKey:   "vhost",
+								DescriptorValue: params.vHost,
+							},
+						},
+					},
+					{
+						ActionSpecifier: &routev3.RateLimit_Action_GenericKey_{
+							GenericKey: &routev3.RateLimit_Action_GenericKey{
+								DescriptorKey:   "path",
+								DescriptorValue: basePath,
+							},
+						},
+					},
+				},
+			}
+			for _, method := range resourceMethods {
+				operationLevelRateLimitActions := []*routev3.RateLimit_Action{
+					{
+						ActionSpecifier: &routev3.RateLimit_Action_GenericKey_{
+							GenericKey: &routev3.RateLimit_Action_GenericKey{
+								DescriptorKey:   "method",
+								DescriptorValue: method,
```

Comment:
If
- API level -> value should be "ALL"
- Operation level -> HTTP method of operation